arrow/src/main/java/org/apache/calcite/adapter/arrow/ArrowTranslator.java

```diff
@@ -185,6 +187,34 @@ private String translateOp2(String op, String name, RexLiteral right) {
     return name + " " + op + " " + valueString + " " + valueType;
   }
 
+  /** Translates a call to a unary operator. */
+  private String translateUnary(String op, RexCall call) {
+    final RexNode opNode = call.operands.get(0);
+    @Nullable String expression = translateUnary2(op, opNode);
+
+    if (expression != null) {
+      return expression;
+    }
+
+    throw new UnsupportedOperationException("Unsupported unary operator " + call);
+  }
+
+  /** Translates a call to a unary operator. Returns null on failure. */
+  private @Nullable String translateUnary2(String op, RexNode opNode) {
+    if (opNode.getKind() == SqlKind.INPUT_REF) {
+      final RexInputRef inputRef = (RexInputRef) opNode;
```

Comment:
Are you sure this conversion will be successful? Do you need to make a judgment?

Reply:
I checked other places, where the same cast happens:
- [Cast in DruidExpressions](https://github.com/apache/calcite/blob/7d9b639cd41ff7b3e3c81a705da5a38308c0b8ac/druid/src/main/java/org/apache/calcite/adapter/druid/DruidExpressions.java#L109-L110)
- [Cast in DruidQuery](https://github.com/apache/calcite/blob/7d9b639cd41ff7b3e3c81a705da5a38308c0b8ac/druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java#L403-L404)
- [Cast in GeodeFilter](https://github.com/apache/calcite/blob/7d9b639cd41ff7b3e3c81a705da5a38308c0b8ac/geode/src/main/java/org/apache/calcite/adapter/geode/rel/GeodeFilter.java#L372-L374)
- [Casts in ProjectAggregateMergeRule](https://github.com/apache/calcite/blob/7d9b639cd41ff7b3e3c81a705da5a38308c0b8ac/core/src/main/java/org/apache/calcite/rel/rules/ProjectAggregateMergeRule.java#L95-L106)

I can add an `instanceof` check, but this might be unnecessary? At least that's the impression I get from the examples above.

Reply:
I don't think it's necessary, as you have noticed it's a common pattern to rely on `SqlKind` to infer the type of an object instead of `instanceof`, for me it's good as-is

Reply:
@asolimando thank you for the explanation